stdlib/http/src/main/java/org/ballerinalang/net/http/WebSocketClientHandshakeListener.java

```diff
@@ -20,31 +20,39 @@
 
 import org.ballerinalang.jvm.BallerinaValues;
 import org.ballerinalang.jvm.values.ObjectValue;
-import org.wso2.transport.http.netty.contract.websocket.ClientHandshakeListener;
+import org.ballerinalang.net.http.websocketclientendpoint.RetryContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.wso2.transport.http.netty.contract.websocket.WebSocketConnection;
 import org.wso2.transport.http.netty.message.HttpCarbonResponse;
 
+import java.io.IOException;
 import java.util.concurrent.CountDownLatch;
 
 import static org.ballerinalang.net.http.HttpConstants.PROTOCOL_HTTP_PKG_ID;
+import static org.ballerinalang.net.http.WebSocketConstants.CONNECTED_TO;
+import static org.ballerinalang.net.http.WebSocketConstants.RETRY_CONFIG;
+import static org.ballerinalang.net.http.WebSocketUtil.hasRetryConfig;
+import static org.ballerinalang.net.http.WebSocketUtil.reconnect;
 
 /**
  * The handshake listener for the client.
  *
  * @since 0.983.1
  */
-public class WebSocketClientHandshakeListener implements ClientHandshakeListener {
+public class WebSocketClientHandshakeListener extends WebSocketClientHandshakeConnectorListener {
```

Comment:
The purpose of inheritance is reuse and reducing duplicates. I don't see that there's much gain by inheritance. Between can we choose better names for super classes?